project/apps/matching-service/src/db/match.redis.ts

```diff
@@ -283,4 +291,42 @@ export class MatchRedis {
       success: true,
     };
   }
+
+  /**
+   * Adds a match_req_Id to the user match map in redis
+   * @param userId The userId to map
+   * @param match_req_id The match_req_id to map
+   */
+
+  async addUserMatchMapaping(userId: string, match_req_id: string) {
+    try {
+      const key = `${MATCH_USER}-${userId}`;
+      await this.redisClient.set(key, match_req_id);
+    } catch (error) {
+      this.logger.error(`Error setting userId to matchId: ${error}`);
+    }
+  }
+
+  /**
+   * Removes a user match mapping
+   * @param userId The userId to map
+   */
+
+  async removeUserMatchMapaping(userId: string) {
+    try {
+      const key = `${MATCH_USER}-${userId}`;
+      await this.redisClient.del(key);
```

Comment:
nit: should aim to log any db write operations like delete
`this.logger.log(`Removed userMatchMapping for user ...`